sceptre/stack.py

```diff
@@ -126,14 +143,25 @@ class Stack(object):
     template_bucket_name = ResolvableValueProperty(
         "template_bucket_name", PlaceholderType.none
     )
-    # Similarly, the placeholder_override=None for iam_role means that actions that would otherwise
-    # use the iam_role will act as if there was no iam role when the iam_role stack has not been
+    # Similarly, the placeholder_override=None for sceptre_role means that actions that would otherwise
+    # use the sceptre_role will act as if there was no iam role when the sceptre_role stack has not been
     # deployed for commands that allow placeholders (like validate).
-    iam_role = ResolvableValueProperty("iam_role", PlaceholderType.none)
-    role_arn = ResolvableValueProperty("role_arn")
+    sceptre_role = ResolvableValueProperty("sceptre_role", PlaceholderType.none)
+    cloudformation_service_role = ResolvableValueProperty("cloudformation_service_role")
 
     hooks = HookProperty("hooks")
 
+    iam_role = create_deprecated_alias_property(
+        "iam_role", "sceptre_role", "4.0.0", "5.0.0"
+    )
+    role_arn = create_deprecated_alias_property(
+        "role_arn", "cloudformation_service_role", "4.0.0", "5.0.0"
+    )
+    sceptre_role_session_duration = None
+    iam_role_session_duration = create_deprecated_alias_property(
+        "iam_role_session_duration", "sceptre_role_session_duration", "4.0.0", "5.0.0"
+    )
+
```

Comment:
These "deprecated alias properties" emit deprecation warnings and ultimately resolve to the correct values, both for getting and setting. You can find this function here: https://github.com/Sceptre/sceptre/pull/1295/files#diff-590afffaaf0b4f3a07656251e95e9b01e40c65fe8b5f8a9bf6cd84f4d3a1e5c4

Reply:
That's pretty slick.